### ISSUE TITLE
0.4.5 python3-catkin-pkg-modules uninstallable on Trusty

### ISSUE THREAD

**Issue by tfoote** [ros-infrastructure/catkin_pkg#227]
Following up https://github.com/ros-infrastructure/ros_buildfarm/issues/559

Here's a log of trying to install it in a clean docker container
https://gist.github.com/tfoote/5d8c0fd497d2dbbdee19c26d083e11e1

**Comment by sloretz** [comment]
Looks like https://bugs.launchpad.net/ubuntu/+source/dpkg/+bug/1730627 . Someone at
[aptly-dev/aptly/#655](https://github.com/aptly-dev/aptly/issues/655) traced it to [dkpg-deb making `--uniform-compression` the default](https://lists.debian.org/debian-devel/2016/07/msg00114.html). It looks like a fix was [released to Trusty on 2018-03-14](https://bugs.launchpad.net/ubuntu/+source/dpkg/+bug/1730627/comments/19).

The `.deb` from the zip archive appears to install fine in up-to-date trusty docker container.

```
$ sudo apt-get install python3-dateutil python3-docutils python3-pyparsing
# ... edited out ...
$ sudo dpkg -i python3-catkin-pkg-modules_0.4.5-1_all.deb
Selecting previously unselected package python3-catkin-pkg-modules.
(Reading database ... 26122 files and directories currently installed.)
Preparing to unpack python3-catkin-pkg-modules_0.4.5-1_all.deb ...
Unpacking python3-catkin-pkg-modules (0.4.5-1) ...
Setting up python3-catkin-pkg-modules (0.4.5-1) ...
$ python3
Python 3.4.3 (default, Nov 28 2017, 16:41:13) 
[GCC 4.8.4] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import catkin_pkg
>>> 
```

If asking users to upgrade `dpkg` on trusty is acceptable then I think this can be closed.

**Comment by tfoote** [comment]
Since trusty is still supported and that's been available I think it's reasonable to do.

We'll also need to rebuild our base images for testing with the new version included. That explains why this passed for amd64, but not i386 and armhf since those are our slightly less updated targets. 

@nuclearsandwich It looks like we should rebuild our base images.

**Comment by nuclearsandwich** [comment]
> @nuclearsandwich It looks like we should rebuild our base images.

New images have been pushed after testing that the i386 and armhf trusty images can install python3-catkin-pkg-modules successfully.

**Comment by dirk-thomas** [comment]
I will go ahead and close this ticket since it sounds like it has been addressed by updating the system.

**Comment by elahia** [comment]
Hi there,
I have the same problem, but by reading the answers, i became more confused! 
would you please clear the final solution?!
and my preference is to avoid installing python3!


Regards,

**Comment by dirk-thomas** [comment]
Please see the comment in the referenced ticket mentioning the exact steps to update `dpkg`: https://github.com/ros-infrastructure/catkin_pkg/issues/230#issuecomment-408913341

**Comment by elahia** [comment]
I did:

```
1. apt-get update
2. apt-get install dpkg
```
but the output is look like this 
![untitled3](https://user-images.githubusercontent.com/14091096/43592403-c53eea5c-9675-11e8-95f2-1753664185b2.png)


 then I tried for  apt-get install python-catkin-pkg, the result is:

```
root@DESKTOP-IQU4A6G:/# apt-get install python-catkin-pkg
Reading package lists... Done
Building dependency tree
Reading state information... Done
You might want to run 'apt-get -f install' to correct these:
The following packages have unmet dependencies:
 python-catkin-pkg : Depends: python-catkin-pkg-modules (>= 0.4.7) but it is not going to be installed
 python-rosdistro-modules : Depends: python-catkin-pkg-modules but it is not going to be installed
E: Unmet dependencies. Try 'apt-get -f install' with no packages (or specify a solution).
root@DESKTOP-IQU4A6G:/#
```

trying for apt-get -f install does not help!

Am I right?

**Comment by sloretz** [comment]
@r1b4z01d https://bugs.launchpad.net/ubuntu/+source/dpkg/+bug/1730627/comments/19 says it's fixed in `1.17.5ubuntu5.8`
